packages/playwright/src/worker/workerMain.ts

```diff
@@ -186,28 +193,43 @@ export class WorkerMain extends ProcessRunner {
       void this._stop();
   }
 
-  private async _loadIfNeeded() {
+  private async _load(): Promise<{
+    config: FullConfigInternal,
+    project: FullProjectInternal,
+    poolBuilder: PoolBuilder,
+  } | undefined> {
     if (this._config)
-      return;
+      return { config: this._config, project: this._project, poolBuilder: this._poolBuilder };
 
     this._config = await deserializeConfig(this._params.config);
-    this._project = this._config.projects.find(p => p.id === this._params.projectId)!;
+    const project = this._config.projects.find(p => p.id === this._params.projectId);
+    if (!project)
+      return undefined;
```

Comment:
After reading through this PR, I realized the following. If we just throw a nicely formatted error here, it will end up as a `fatalError` and will be presented to the user in a nice way and cancel any tests in this run. The only problem is we'll probably have multiple of them - one per test group, but that's not a big deal for such an edge case. WDYT?